static/app/components/replays/replayContext.tsx

```diff
@@ -374,6 +394,31 @@ export function Provider({
         }
       }
 
+      // check if this is a video replay and if we can use the video replayer
+      if (isVideoReplay && videoAttachments && startTimestampMs) {
+        const inst = new VideoReplayer(videoAttachments, {
+          videoApiPrefix: `/api/0/projects/${
+            organization.slug
+          }/${projectSlug}/replays/${replay?.getReplay().id}/videos/`,
+          root,
+          start: startTimestampMs,
+          onFinished: setReplayFinished,
+          onLoaded: event => {
+            forceDimensions({
+              height: event.target.videoHeight,
+              width: event.target.videoWidth,
+            });
+          },
+        });
+        // `.current` is marked as readonly, but it's safe to set the value from
+        // inside a `useEffect` hook.
+        // See: https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables
+        // @ts-expect-error
+        replayerRef.current = inst;
+        applyInitialOffset();
+        return;
+      }
+
```

Comment:
Where the magic happens... `VideoReplayer` should have the same interface as `Replayer`